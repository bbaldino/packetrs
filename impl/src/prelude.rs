pub use crate::{
    bail, bitcursor::*, error::PacketRsResult, packetrs_read::PacketRsRead, types::*, Context,
    Result,
};
