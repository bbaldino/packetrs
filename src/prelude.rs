pub use crate::{
    anyhow::*, 
    error,
    packetrs_read::PacketrsRead, ux::*, PacketrsRead,
};

pub use packetrs_impl::b4::{bitvec, bit_cursor::BitCursor, bit_read::BitRead, bit_read_exts::BitReadExts, bit_vec::BitVec, byte_order::*};
