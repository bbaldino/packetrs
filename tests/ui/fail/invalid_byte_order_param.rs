use packetrs::prelude::*;

#[derive(PacketrsRead)]
#[packetrs(byte_order = "blah")]
struct Foo {
}

fn main() {}
