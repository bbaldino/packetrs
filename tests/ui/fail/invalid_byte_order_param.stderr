error: proc-macro derive panicked
 --> tests/ui/fail/invalid_byte_order_param.rs:3:10
  |
3 | #[derive(PacketrsRead)]
  |          ^^^^^^^^^^^^
  |
  = help: message: Error: 'byte_order' parameter is invalid
